File: 005-TypeScript/002-Orintacao a Objeto/src/calculadora.ts
export class Calculadora{
    //Modificadores de Acesso: public, private, protected, readonly 
    private memoria: number; 

    constructor(memoria: number){
        this.memoria = memoria; 
    }

    public somar(num1: number, num2: number): number{
        return num1 + num2; 
    }

    public subtrair(num1: number, num2: number): number{
        return num1 - num2; 
    }

    public dividir(num1: number, num2: number): number{
        return num1 / num2; 
    }
   
    public multiplicar(num1: number, num2: number): number{
        return num1 * num2; 
    }


    public getMemoria():number{
        return this.memoria; 
    }

    public setMemoria(num: number):void{
        this.memoria = num; 
    }

}

File: 008-Java/000 - Exercícios/Lista3E1/src/Calculadora.java

public class Calculadora {

	private double memoria; 
	
	
	public double somar(double num1, double num2) {
		return num1 + num2; 
	}
	
	public double subtrair(double num1, double num2) {
		return num1 - num2; 
	}
	
	public double multiplicar(double num1, double num2) {
		return num1 * num2; 
	}
	
	public double dividir(double num1, double num2) {
		return num1 / num2; 
	}
	
	public void addMemoria(double num1) {
		this.memoria = num1; 
	}
	
	public void clrMemoria() {
		this.memoria = 0; 
	}
	
	public double getMemoria() {
		return this.memoria;
	}
	
}
